src/full/Agda/TypeChecking/Rewriting.hs

```diff
@@ -363,7 +363,7 @@ checkRewriteRule q = do
     checkAxFunOrCon :: QName -> Definition -> TCM ()
     checkAxFunOrCon f def = case theDef def of
       Axiom{}        -> return ()
-      def@Function{} -> whenJust (funProjection def) $ \proj ->
+      def@Function{} -> flip (either (const (pure ()))) (funProjection def) $ \proj ->
```

Comment:
Consider adding a combinator
```haskell
whenRight :: Monad m => Either a b -> (b -> m ()) -> m ()
```
to `Agda.Utils.Either` and use it here.  
Actually `whenRight = whenJust . maybeRight`, so consider:
```haskell
whenJust (maybeRight (funProjection def)) $ \ proj -> 
```